src/WeakMap.php

```diff
@@ -88,7 +88,6 @@ public function offsetExists($object) : bool
 
         public function offsetGet($object)
         {
-            $this->housekeeping();
```

Comment:
> no housekeeping on get as it can possibly trigger GC

Can you please elaborate on this one? Adding it back does not seem to break any tests.

Reply:
`housekeeping` method increments `housekeepingCounter`, we must assure when `offsetExists()` returns true, `offsetGet()` will return the object and never throw

Reply:
I have reverted this as the key is usually not unreferenced between isset/get.